File: src/main/java/core/parser/processor/string/DigitFilter.java
package core.parser.processor.string;

import core.parser.processor.IProcessor;

public class DigitFilter implements IProcessor<String> {
    @Override
    public String process(String in) {
        String cleaned = in.trim().replaceAll("\\W+", "");
        return (cleaned.matches("\\d+")) ? "" : in;
    }
}

File: src/main/java/core/parser/processor/string/DigitFilter.kt
package core.parser.processor.string

import core.parser.processor.IProcessor
import org.apache.commons.lang3.math.NumberUtils

class DigitFilter : IProcessor<String> {
    override fun process(`in`: String): String {
        if (NumberUtils.isDigits(`in`))
            return ""
        else
            return `in`
    }
}
